packages/walletconnect-v1/src/wc-wallet.ts

```diff
@@ -98,17 +96,38 @@ export class WalletConnectWallet extends RemoteWallet<WalletConnectSigner> {
       // create new session
       await this.client.createSession(this.connectOptions);
     }
+  }
+
+  async switchToChain(params: {
+    chainId: number;
+    networkId: number;
+    rpcUrl: string;
+    nativeCurrency: { name: string; symbol: string };
+  }) {
+    return this.client!.updateChain(params);
+  }
+
+  /**
+   * Get the URI needed for out of band session establishment
+   */
+  public async getUri() {
+    await this.setupClient();
 
     return this.client?.uri;
   }
 
-  onSessionCreated = (error: Error | null, session: IInternalEvent): void => {
+  // heads up! common pattern for the onSession*** methods is to overwrite them externally
```

Comment:
don't we want to get rid of that ?

Reply:
maybe but i can only do so much

Reply:
i was worried it would cause breaking changes to that api